js/runtime.js

```diff
@@ -821,51 +823,165 @@
+                        break;
+                    case "1/16":
+                        length = "16";
+                        break;
+                    case "1/8":
+                        length = "8";
+                        break;
+                    case "1/4":
+                        length = "4";
+                        break;
+                    case "1/2":
+                        length = "2";
+                        break;
+                    case "1":
+                        length = "1";
+                        break;
+                }
+                var newNote = note + length;
+                song += newNote;
+            },
+            playAudio: function(audio){
+                audio.play();
+            },
+            playInstrument: function(sound){
+                console.log(song);
+                T("mml", {mml:song}, sound).on("ended", function() {
+                    sound.pause();
+                    this.stop();
+                }).start();
+                song = "o4 l4 V12 ";
+            },
+            mml: function(sound, mml){
+                var gen = T("OscGen", {wave:sound, env:{type:"perc"}, mul:0.25}).play();
+                T("mml", {mml:mml}, gen).on("ended", function() {
+                    gen.pause();
+                    this.stop();
+                }).start();
+            },
+            tempo: function(tempo){
+                song += ("t" + tempo + " ");
             },
             pause: function(sound){
                 sound.pause();
             },
-            playFrom: function(sound, time){
-                sound.playFrom(time);
-            },
-            pan: function(sound, balance){
-                sound.pan = balance;
-            },
-            echo_DelayFeedbackFilter: function(sound, delay, feedback, filter){
-                sound.setEcho(delay, feedback, filter);
-            },
-            stopEcho: function(sound){
-                sound.echo = false;
-            },
-            reverb_DurationDecayReverse: function(sound, duration, decay, reverse){
-                sound.setReverb(duration, decay, reverse);
-            },
-            stopReverb: function(sound){
-                sound.reverb = false;
+            keys: function(wave, vol){
+                var synth = T("OscGen", {wave:wave, mul:vol}).play();
+
+                var keydict = T("ndict.key");
+                var midicps = T("midicps");
+                T("keyboard").on("keydown", function(e) {
+                  var midi = keydict.at(e.keyCode);
+                  if (midi) {
+                    var freq = midicps.at(midi);
+                    synth.noteOnWithFreq(freq, 100);
+                  }
+                }).on("keyup", function(e) {
+                  var midi = keydict.at(e.keyCode);
+                  if (midi) {
+                    synth.noteOff(midi, 100);
+                  }
+                }).start();
+
+                alert("Play notes on the keyboard");
```

Comment:
Yeah it's intended to notify users that the keyboard is now playable. Though I suppose it would just be annoying if someone already knows that.